### DISCUSSION TITLE
I am horrified

### DISCUSSION THREAD

**Discussion by Immons** [dotnet/maui#10431] [General]
Honestly, I am horrified, I feel like I stepped back 8 years and again trying to do something in Xamarin.Forms from 2014. Maui is not ready for development, half features are not working and migrating bigger projects would take weeks. If you really plan to kill Xamarin.Forms and make us use that half-made product, then you will probably lose more developers than gain new ones.

**Comment by xtuzy** [comment]
Xamarin.Forms will be supported before 202405，MAUI will have 2 year become well,  if now MAUI not well, why you want migrate xamarinform project immediately.

**Comment by AnthonyH7386** [comment]
> Honestly, I am horrified, I feel like I stepped back 8 years and again trying to do something in Xamarin.Forms from 2014. Maui is not ready for development, half features are not working and migrating bigger projects would take weeks. If you really plan to kill Xamarin.Forms and make us use that half-made product, then you will probably lose more developers than gain new ones.

I have to agree with you.  Maui is sluggish and jittery compared to Xamarin Forms.  They can’t even get the keyboard functionality right!  Buttons don’t work, it stays on screen when navigating between pages and there’s a horrible big black hole underneath it when it finally animates off screen!

There’s a lot of work to be done before I’ll even consider migrating my projects across.  Extremely disappointing!
